lib/class-wp-theme-json.php

```diff
@@ -1127,31 +1127,48 @@ public function get_stylesheet( $type = 'all' ) {
 	 * @param WP_Theme_JSON $incoming Data to merge.
 	 */
 	public function merge( $incoming ) {
-		$incoming_data    = $incoming->get_raw_data();
-		$this->theme_json = array_replace_recursive( $this->theme_json, $incoming_data );
+		$incoming_data = $incoming->get_raw_data();
+		$existing_data = $this->theme_json;
 
 		// The array_replace_recursive algorithm merges at the leaf level.
 		// For leaf values that are arrays it will use the numeric indexes for replacement.
-		// In those cases, what we want is to use the incoming value, if it exists.
-		//
-		// These are the cases that have array values at the leaf levels.
-		$properties   = array();
-		$properties[] = array( 'color', 'palette' );
-		$properties[] = array( 'color', 'gradients' );
-		$properties[] = array( 'custom' );
-		$properties[] = array( 'spacing', 'units' );
-		$properties[] = array( 'typography', 'fontSizes' );
-		$properties[] = array( 'typography', 'fontFamilies' );
+		$this->theme_json = array_replace_recursive( $this->theme_json, $incoming_data );
+
+		// There are a few cases in which we want to merge things differently
+		// from what array_replace_recursive does.
+
+		// Some incoming properties should replace the existing.
+		$to_replace   = array();
+		$to_replace[] = array( 'custom' );
+		$to_replace[] = array( 'spacing', 'units' );
+		$to_replace[] = array( 'typography', 'fontSizes' );
+		$to_replace[] = array( 'typography', 'fontFamilies' );
+
+		// Some others should be appended to the existing.
+		$to_append   = array();
+		$to_append[] = array( 'color', 'palette' );
+		$to_append[] = array( 'color', 'gradients' );
```

Comment:
Already existing issue: It seems we need to include array( 'color', 'duotone' ) in either 
 to_append or to_replace.

Reply:
Did the same as the other color palettes: core is always enqueued and the theme is appended.